pkgs/development/tools/godot/default.nix

```diff
@@ -65,39 +83,76 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  sconsFlags = "target=release_debug platform=x11";
-  preConfigure = ''
-    sconsFlags+=" ${
-      lib.concatStringsSep " "
-      (lib.mapAttrsToList (k: v: "${k}=${builtins.toJSON v}") options)
-    }"
-  '';
+  sconsFlags = let toSconsValue = x: if builtins.isBool x then lib.boolToString x else x; in
+    lib.pipe self.passthru.sconsFlagSet [
+      (lib.mapAttrsToList (k: v: "${k}=${toSconsValue v}"))
+      (lib.concatStringsSep " ")
+    ];
```

Comment:
I'd imagine this could be greatly simplified

Reply:
Any suggestions?

Reply:
Use strings directly instead of lists of types

Reply:
Where are there lists of types?  I'm not sure what you mean...

Reply:
Oh, do you mean have `sconsFlagSet` require all the attribute values to already be strings, so that we don't have to use `toSconsValue` to convert them into strings?  If so, that would simplify the logic for determining `sconsFlags`, but it would be less ergonomic for when the `sconsFlagSet` is actually defined, and I'd rather favor the simplicity of usage over the simplicity of implementation.  I'll see if I can think of a way to simplify this, though, but not sure if I can.